pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -126,18 +112,18 @@ func (e *edgeFetcher) updateWatermark(inputOffset isb.Offset, fromPartitionIdx i
 	e.lastProcessedWm[fromPartitionIdx] = epoch
 	e.Unlock()
 
-	e.log.Debugf("%s[%s] processed watermark for offset %d: %+v", debugString.String(), e.bucketName, offset, epoch)
-	return nil
+	e.log.Debugf("%s processed watermark for offset %d: %+v", debugString.String(), offset, epoch)
+	return wmb.Watermark(time.UnixMilli(epoch))
 }
 
-// GetHeadWatermark returns the smallest head watermark among all the processors for given partition
+// ComputeHeadWatermark returns the smallest head watermark among all the processors for given partition
 // This can be used in showing the watermark
 // progression for a vertex when not consuming the messages directly (eg. UX, tests)
 // NOTE
 //   - We don't use this function in the regular pods in the vertex.
-//   - UX only uses GetHeadWatermark, so the `p.IsDeleted()` check in updateWatermark() never happens.
+//   - UX only uses ComputeHeadWatermark, so the `p.IsDeleted()` check in updateWatermark() never happens.
 //     Meaning, in the UX (daemon service) we never delete any processor.
-func (e *edgeFetcher) GetHeadWatermark(fromPartitionIdx int32) wmb.Watermark {
+func (e *edgeFetcher) ComputeHeadWatermark(fromPartitionIdx int32) wmb.Watermark {
```

Comment:
```suggestion
func (e *edgeFetcher) ComputeHeadWatermarkForUX(fromPartitionIdx int32) wmb.Watermark {
```

since this is used only by UX?

Reply:
hmmmm the function itself is a method of the UXFetcher interface... so I didn't add the UX to the function name
```
// UXFetcher computes the watermark for Vn.
type UXFetcher interface {
	...
	// ComputeHeadWatermark computes a valid head watermark for the given partition
	ComputeHeadWatermark(fromPartitionIdx int32) wmb.Watermark
}
```